firstvoices/backend/tasks/import_job_tasks.py

```diff
@@ -89,12 +96,42 @@ def clean_csv(data):
 
     # Dropping invalid columns
     for invalid_header in invalid_headers:
-        del data[invalid_header]
+        del cleaned_data[invalid_header]
 
     # lower-casing headers
-    data.headers = [header.lower() for header in data.headers]
-
-    return accepted_headers, invalid_headers, data
+    cleaned_data.headers = [header.lower() for header in cleaned_data.headers]
+
+    return accepted_headers, invalid_headers, cleaned_data
+
+
+def get_failed_rows_csv_file(import_job_instance, data, error_row_numbers):
+    # Generate a csv for the erroneous rows
+    failed_row_dataset = []
+    for row_num in error_row_numbers:
+        failed_row_dataset.append(
+            data[row_num - 1]
+        )  # -1 to subtract to account for headers
+    failed_row_dataset = tablib.Dataset(*failed_row_dataset)
+    failed_row_dataset.headers = data.headers
+
+    failed_row_export = failed_row_dataset.export("csv")
+    in_memory_csv_file = io.BytesIO(failed_row_export.encode("utf-8"))
+    in_memory_csv_file = InMemoryUploadedFile(
+        file=in_memory_csv_file,
+        field_name="failed_rows_csv",
+        name="failed_rows.csv",
+        content_type="text/csv",
+        size=sys.getsizeof(in_memory_csv_file),
+        charset="utf-8",
+    )
+    failed_row_csv_file = File(
+        content=in_memory_csv_file,
+        site=import_job_instance.site,
+        created_by=import_job_instance.created_by,
+        last_modified_by=import_job_instance.last_modified_by,
```

Comment:
Totally minor, but I think the created_by and last_modified_by here should both be set to the import_job_instance.last_modified_by value just in case there is any difference